### ISSUE TITLE
convert table to dataframe (and back again)

### ISSUE THREAD

**Issue by sdmcallister** [JuliaComputing/JuliaDB.jl#95]
Perhaps this function could be improved on (or does one exist that I missed?):
```
function DataFrame(t)
       df = DataFrame()
       for i in colnames(t)
           df[i]=collect(select(t,i))
       end
       df
end
```
It is a pretty trivial function, but having something that converts a table to a dataframe with very little typing is handy. Probably be pretty easy to add a table(df::DataFrame) function as well.

Thoughts?

**Comment by shashi** [comment]
If you use Query you can already do this...

Maybe `DataStreams` integration is now possible. cc @quinnj, would that allow this?

**Comment by quinnj** [comment]
Yes, and it could be very fast too w/ even non-copying of columns. JuliaDB support is high on my DataStreams todo list.

**Comment by davidanthoff** [comment]
It should be enough to load ``IterableTables``, no need for ``Query`` and then it should all just work.

**Comment by shashi** [comment]
@davidanthoff 

```julia
julia> table(dataset("datasets", "iris"))
ERROR: ArgumentError: iter cannot be turned into a NextTable.
Stacktrace:
 [1] #table#289(::Array{Any,1}, ::Function, ::DataFrames.DataFrame) at /home/shashi/.julia/v0.6/IndexedTables/src/tabletraits.jl:68
 [2] table(::DataFrames.DataFrame) at /home/shashi/.julia/v0.6/IndexedTables/src/tabletraits.jl:60
 [3] macro expansion at /home/shashi/.julia/v0.6/Revise/src/Revise.jl:775 [inlined]
 [4] (::Revise.##17#18{Base.REPL.REPLBackend})() at ./event.jl:73
```
I thought `isiterabletable` should return `true` for DataFrame?

**Comment by davidanthoff** [comment]
You need to add a ``import IterableTables`` somewhere because the iterable tables integration for ``DataFrame`` is not part of DataFrames.jl but in IterableTables.jl. Hopefully at some point we can move it into DataFrames.jl itself (the way it is in IndexedTables.jl) and IterableTables.jl can disappear.

**Comment by quinnj** [comment]
Actually, we shouldn't need anything in IterableTables for DataFrames since it implements `Data.Source` interface from DataStreams.jl. We can just use `Data.rows(df)` to get a type stable row iterator of NamedTuples (works on both 0.6 and 0.7).

**Comment by davidanthoff** [comment]
The iterator you get back from ``Data.rows(df)`` is not compatible with the iterable tables ecosystem because it uses ``Missing`` and not ``DataValue`` for missing values. If we find a solution for the Query & Missing situation, we would change the iterable tables standard, but the earliest I see that happening would be julia 1.1 (if additional compiler optimizations for the 2^n return type explosion in a ``map`` are added by then). You also need IterableTables.jl for the conversion of IndexedTables -> DataFrame, as far as I can tell. The proper solution for all of this would simply be to move the iterable tables integration for DataFrames from IterableTables.jl into DataFrames.jl (like it is done here in IndexedTables.jl).
